### PR TITLE
Extend documentation and move to precice.org

### LINKED ISSUE #2: Refactor documentation and render on precice.org
Let's create a page in the preCICE docs under `Tooling` named "FMI Runner". There, we want to render documentation (could be a single page), which is, however, hosted in this repo. Compare the OpenFOAM adapter or, very recently, the Micro Manager on how this works technically, i.e. what needs to be created on the website.

https://github.com/precice/precice.github.io/pull/270/files

In this repo here, we should the actual documentation page to a subfolder `docs`. The main README should then only contain minimal information ("what is this", "where to find out more") and link to the website. Also, content from other README pages could be ported to the docs.

### PR DESCRIPTION
**Pull request by LeonardWilleke**
closes #2 
- Add a configuration example and citation template
- Move content to new subfolder docs/
- Add workflow `update-website.yml` to update the website automatically
- Move to a branching workflow with `main` and `develop`

A second PR to precice.org is open: https://github.com/precice/precice.github.io/pull/277 
I understood from the discussion in the [Micro Manager update](https://github.com/precice/precice.github.io/pull/270/files) that we have to merge the changes in the FMI Runner repo first, so that the website finds the content at the correct place. Is that correct?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,15 @@
 # preCICE-FMI Runner
 
-The [Functional Mock-Up Interface](https://fmi-standard.org/) (FMI) is a standard for the exchange of dynamic simulation models. Currently, it is the de-facto industry standard for co-simulation. The models implementing the FMI standard are called Functional Mock-Up Units (FMU).
+A tool to couple FMU models with other simulation tools via the coupling library [preCICE](https://precice.org/).
 
-This project contains the runner script `fmiprecice` to couple FMU models with other simulation tools via the coupling library [preCICE](https://precice.org/). The runner serves as an importer for the FMU to steer the simulation. Additionally, it calls the preCICE library to communicate and coordinate with other solvers. 
+## Start here
```
Comment (uekerman) [inline README.md]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -1,48 +1,15 @@
...
 
-![img](images/precice-fmi-runner-setup.png)
+The main documentation is on the preCICE website **add final link**.
```
Comment (uekerman) [inline README.md]:
```suggestion
The main documentation is rendered on the [preCICE website](https://precice.org/tooling-fmi-runner.html).
```

File: README.md [hunk 3]
```diff
@@ -1,48 +1,15 @@
...
-## Usage
-
-The Runner is called from the terminal with the command `fmiprecice`. It takes two input files, one with settings for the FMU simulation and one with settings for preCICE. Start the runner by pointing to the input files:
-
-```bash
-fmiprecice ./fmi-settings.json ./precice-settings.json
-```
-
-You can find more information about the runner, its abilities, and its limitations in [1]. This adaption of the [oscillator tutorial](https://github.com/LeonardWilleke/precice-tutorials/tree/create-fmu-oscillator-v2/oscillator) features a first complete setup with FMU model and setting files. 
+Please report any bugs or issues [here](https://github.com/precice/fmi-runner/issues) and give us feedback through [one of our community channels](https://precice.org/community-channels.html).
```
Comment (uekerman) [inline README.md]:
```suggestion
Please report [issues](https://github.com/precice/fmi-runner/issues) and give us feedback through [one of our community channels](https://precice.org/community-channels.html).
```

File: README.md [hunk 4]
```diff
@@ -1,48 +1,15 @@
...
 
 ## References
 
-[1] L. Willeke, A preCICE-FMI Runner to couple controller models to PDEs, Master Thesis, University of Stuttgart, 2023 (Publication pending)
+The concept and design of the FMI runner is described in:
 
+L. Willeke, [A preCICE-FMI Runner to couple controller models to PDEs]( http://dx.doi.org/10.18419/opus-13130), Master Thesis, University of Stuttgart, 2023
```
Comment (uekerman) [inline README.md]:
```suggestion
L. Willeke, [A preCICE-FMI Runner to couple controller models to PDEs](http://dx.doi.org/10.18419/opus-13130), Master Thesis, University of Stuttgart, 2023
```

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
+---
+title: The FMI runner
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
title: FMI runner
```

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+permalink: tooling-fmi-runner.html
+keywords: tooling, fmi, runner, fmi-settings, precice-settings
+summary: A tool to couple FMUs to PDE-based solvers using preCICE.
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
summary: A tool to execute FMUs and couple them to PDE-based solvers using preCICE.
```

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+---
+
+## What is this?
+
+The [Functional Mock-Up Interface](https://fmi-standard.org/) (FMI) is a standard for the exchange of dynamic simulation models. Currently, it is the de-facto industry standard for co-simulation. The models implementing the FMI standard are called Functional Mock-Up Units (FMU).
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
The [Functional Mock-Up Interface](https://fmi-standard.org/) (FMI) is a standard for the exchange of dynamic simulation models. Currently, it is the de-facto industry standard for co-simulation. Models implementing the FMI standard are called Functional Mock-Up Units (FMU).
```

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+This tool contains the runner script `fmiprecice` to couple FMU models with other simulation tools via the coupling library [preCICE](https://precice.org/). The runner serves as an importer for the FMU to steer the simulation. Additionally, it calls the preCICE library to communicate and coordinate with other solvers.
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
The preCICE-FMI runner contains the script `fmiprecice` to couple FMU models with other simulation tools via [preCICE](https://precice.org/). The runner serves as an importer for the FMU to steer the simulation. Additionally, the runner calls the preCICE library to communicate and coordinate with other solvers. 
```

File: docs/README.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+Read on to find out how to install and configure the runner. More information about the software, its abilities, and its limitations can be found in [1]. If you are ready to run your first case, have a look at the [oscillator tutorial](https://github.com/LeonardWilleke/precice-tutorials/tree/create-fmu-oscillator-v2/oscillator). 
+
+## Get the Runner
+
+### Dependencies
+
+Before you start the installation, make sure you have installed the following packages:
+
+* Python 3 or higher
+* [preCICE](https://precice.org/installation-overview.html), v2
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
* [preCICE](https://precice.org/installation-overview.html)
```
anyway clear

File: docs/README.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+
+```bash
+export PATH=$PATH:$HOME/.local/bin
+```
+
+You're good to go! Read on to learn how to configure the FMI runner or [run your first simulation](https://github.com/LeonardWilleke/precice-tutorials/tree/create-fmu-oscillator-v2/oscillator).
+
+## Configuration
+
+We use two settings files to configure the software to the specific simulation case.
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
We use two settings files to configure the software for a specific simulation case.
```

File: docs/README.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    "simulation_params": {
+        ...
+        "fmu_read_data_name": ["force_1", "force_2"],
+        ...
+    }
+```
+
+{% disclaimer %}
+The runner v0.1 can not deal with FMU vectors. All FMU variables need to be scalars. Besides that, any data type implemented in the FMU (integer, float, boolean, string) is supported.
+{% enddisclaimer %}
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
{% note %}
Version v0.1 of the runner cannot deal with FMU vectors. All FMU variables need to be scalars. Besides that, however, any data type implemented in the FMU (integer, float, boolean, string) is supported.
{% endnote %}
```

File: docs/README.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+
+Optionally, you can choose variables in `output` that are tracked and stored as a timeseries in `output_file_name`.
+
+
```
Comment (uekerman) [inline docs/README.md]:
```suggestion

```

File: docs/README.md [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+**model_params**: Use this to set internal parameters before the start of the simulation. In this example, the parameters `apply_filter` and `spring_coeff` are set. The keys of this dictionary are the names of specific FMU variables.
+
+**initial_conditions**: Use this to set initial conditions for internal parameters. Again, the keys of this dictionary are the names of specific FMU variables.
+
+**input_signals**: You can set time-dependent input signals. The list `names` holds the different parameters. The first entry of this list is always `time`. The list `data` sets the input values for the variables given in `time`. The first entry of each nested list in `data` is the time at which the following data is set. Here, `damping_coef` is set to 5.0 at `t=2.0`. Of course, you can also set the values of boolean and string parameters.
+
+### Configure the coupling
+
+The file `precice-settings.json` is used to configure the coupling with preCICE. It consist of one dictionary **coupling_params**.
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
The file `precice-settings.json` is used to configure the coupling with preCICE. It consists of one dictionary **coupling_params**.
```

File: docs/README.md [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+        "participant_name": "Suspension",
+        "config_file_name": "../precice-config.xml",
+        "mesh_name": "Suspension-Mesh",
+        "read_data": {"name": "Force", "type": "scalar"}, 
+        "write_data": {"name": "Position", "type": "scalar"}
+    }
+}
+```
+
+The `participant_name` and `mesh_name` need to match the entries in the `precice-config.xml`. The same is true for `write_data` and `read_data`, which control the data exchange. The `type` can be set to `scalar` or `vector`.
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
`participant_name` and `mesh_name` need to match the entries in `precice-config.xml`. The same is true for `write_data` and `read_data`, which control the data exchange. `type` can be set to `scalar` or `vector`.
```

File: docs/README.md [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+@misc{
+    Willeke:2023,
+    author   = {Willeke, Leonard},
+    title    = {A preCICE-FMI Runner to Couple Controller Models to PDEs},
+    school   = {University of Stuttgart},
+    type     = {Master thesis},
+    doi      = {10.18419/opus-13130},
+    year     = {2023}
+},
+
```
Comment (uekerman) [inline docs/README.md]:
```suggestion
}
```

**Review by uekerman — APPROVED** [review]
Content looks good.

@MakisH Could you please have a quick look on the technical website aspects.

File: docs/README.md [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+## Usage
+
+The runner is called from the terminal with the command `fmiprecice`. It takes two input files, one with settings for the FMU simulation and one with settings for preCICE. Start the runner by pointing to the input files:
+
+```bash
+fmiprecice ./fmi-settings.json ./precice-settings.json
+```
+
+Read on to find out how to install and configure the runner. More information about the software, its abilities, and its limitations can be found in [1]. If you are ready to run your first case, have a look at the [oscillator tutorial](https://github.com/LeonardWilleke/precice-tutorials/tree/create-fmu-oscillator-v2/oscillator).
```
Comment (MakisH) [inline docs/README.md:24]: This link should eventually point to the preCICE website.

File: docs/README.md [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+python setup.py install --user
+```
+
+Maybe you need to add your user installation to your path to make the new software findable. To do so, run the command:
+
+```bash
+export PATH=$PATH:$HOME/.local/bin
+```
+
+You're good to go! Read on to learn how to configure the FMI runner or [run your first simulation](https://github.com/LeonardWilleke/precice-tutorials/tree/create-fmu-oscillator-v2/oscillator).
```
Comment (MakisH) [inline docs/README.md:75]: Also here: point to the website.

File: docs/README.md (hunk 16 above)
Comment (LeonardWilleke) [reply docs/README.md:24]: I have to re-work the FMI runner and the tutorial for preCICE v3 before we bring the FMI tutorial on the website. I would publish the documentation for the runner already and update it with the correct link once the tutorial is finished.

**Pull request merged**